src/harness/vfsUtil.ts

```diff
@@ -1148,7 +1148,7 @@ export class FileSystem {
         for (const key of Object.keys(files)) {
             const value = normalizeFileSetEntry(files[key]);
             const path = dirname ? vpath.resolve(dirname, key) : key;
-            vpath.validate(path, vpath.ValidationFlags.Absolute);
+            vpath.validate(path, vpath.ValidationFlags.Absolute | vpath.ValidationFlags.AllowWildcard);

```

Comment:
I have no idea if this is completely correct. I added it to make [pathCompletionsPackageJsonImportsSrcNoDistWildcard6](https://github.com/microsoft/TypeScript/blob/a80cdfaf5aee37250ae5cfae3d374f8773bd1dd4/tests/cases/fourslash/server/pathCompletionsPackageJsonImportsSrcNoDistWildcard6.ts) pass as it's using `?` in the component path component. 

I have added that test based on the existing [pathCompletionsPackageJsonExportsWildcard6](https://github.com/microsoft/TypeScript/blob/b8e4ed8aeb0b228f544c5736908c31f136a9f7e3/tests/cases/fourslash/pathCompletionsPackageJsonExportsWildcard6.ts). The difference is that the test I have added is a `fourslash/server` test and it follows a slightly different codepath.

The existing one validates here:
https://github.com/microsoft/TypeScript/blob/b8e4ed8aeb0b228f544c5736908c31f136a9f7e3/src/harness/vfsUtil.ts#L1104-L1108

And the `ValidationFlags.AllowWildcard` was added there [here](https://github.com/microsoft/TypeScript/pull/24076/files#diff-effd49d37f377e1f194f7ceaddf609b1d5eb60744ad741cf4e43e11e6878f79dR872-R873)

On the other hand, the `fourslash/server` validates here, where I'm adding this comment. This codepath was added as part of https://github.com/microsoft/TypeScript/pull/20763 - which proceeds the PR that introduced `ValidationFlags.AllowWildcard`. So perhaps this was just a harmless omission in that newer PR. I don't know why this is a flag in the first place though so 🤷‍♂️

Reply:
@rbuckton can you advise?